lib/middleware/connect.js

```diff
@@ -1,24 +1,30 @@
 'use strict';
 
 var Raven = require('../client');
+var utils = require('../utils');
 
 // Legacy support
 var connectMiddleware = function (client) {
   return connectMiddleware.errorHandler(client);
 };
 
+var getClient = function (clientOrDSN) {
+  // Raven is an instance, so use Raven.constructor for instanceof check
+  return clientOrDSN instanceof Raven.constructor ? clientOrDSN : new Raven.Client(clientOrDSN);
+};
+
 // Error handler. This should be the last item listed in middleware, but
 // before any other error handlers.
-connectMiddleware.errorHandler = function (client) {
-  client = client instanceof Raven.Client ? client : new Raven.Client(client);
-  return client.errorHandler();
+connectMiddleware.errorHandler = function (clientOrDSN) {
+  utils.consoleAlert('top-level Raven.middleware.* methods have been deprecated and will be removed in v2.0; use instance methods instead');
```

Comment:
Sort of trying to keep that warning from being excessively long, but I'll be more specific about the methods I'm referring to.

Harm is that a user might just pass their DSN string in 3 places and end up with 3 instances, two of which are sort of hidden and probably entirely unnecessary, when they really just want one instance. Might result in situations where e.g. user expects to get breadcrumbs but doesn't. I want to avoid multiple instances entirely unless the user is explicitly asking for them and knows what they're doing.